docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,171 @@
+1. Depending on the test cases, switchovers will be triggered and verified. 
+1. Control plane status, DPU counters, metering data will be measured for the tests.
+
+## Test Plan
+Assuming there is a pair of DPUs in the system, and at the step of test setup, we have DPU-1 as active, DPU-2 as standby. 
+
+
+### Downstream (T2->T0) Traffic Verification 
+
+For some of the cases in downstream traffic verification, you will see two versions, one is traffic sending through DPU1, and the other one is traffic sending through DPU2. The purpose of  it is to verify:
+* Traffic shouldn’t be disrupted when failures happen on standby side.
+* Traffic should be tunneled if landing on standby side. 
+
+The name convention of a test case will be “\<Test Scenario\>-[Active|Standby]”, indicating the traffic is sent through the initial active or standby side. 
+
+####  Module 1 	Normal OP and Planned Events
+| Case                      | Goal                                       | Test Steps                                                                                 | Expected Control Plane Behavior                               | Expected Data Plane Behavior                      |
+|---------------------------|--------------------------------------------|--------------------------------------------------------------------------------------------|--------------------------------------------------------------|---------------------------------------------------|
+| Normal OP – Active        | Verify normal operation in healthy state  | • Start downstream I/O through Active side                                                 | DPU1 remains active, DPU2 remains standby.                   | T0 receives packets without disruption.          |
+| Normal OP – Standby       | Verify normal operation in healthy state  | • Start downstream I/O through Standby side                                                | DPU1 remains active, DPU2 remains standby.                   | T0 receives packets without disruption.          |
+| Planned Switchover - Active | Verify zero traffic loss in planned maintenance | • Start downstream I/O through active side<br>• Issue switchover following planned maintenance procedure | DPU-1 becomes standby, DPU-2 becomes active.                | T0 receives packets without disruption.          |
+| Planned Switchover – Standby | Verify zero traffic loss in planned maintenance | • Start downstream I/O through standby side<br>• Issue switchover following planned maintenance procedure | DPU-1 becomes standby, DPU-2 becomes active.              | T0 receives packets without disruption.          |
+| Planned shutdown          | Verify zero traffic loss in planned shutdown | • Start downstream I/O <br>• Issue shutdown on standby side                                 | DPU-1 becomes standalone, DPU-2 is dead.                    | T0 receives packets without disruption.          |
```

Comment:
in DPU-driven mode, we can do planned shutdown the active side as well.

Reply:
Addressed.